clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -208,26 +208,26 @@ public static String toHtml() {
             // Responses
             b.append("<b>Responses:</b><br>\n");
             Schema[] responses = key.messageType.responseSchemas();
-            for (int i = 0; i < responses.length; i++) {
-                Schema schema = responses[i];
+            for (int version = key.oldestVersion(); version < key.latestVersion(); version++) {
```

Comment:
Use the same approach we use for requests for consistency - it should result in the same behavior, but fail more clearly if there is some inconsistency.